osmnx/bearing.py

```diff
@@ -195,44 +195,46 @@ def _extract_edge_bearings(
         Ignore edges with `length` attributes less than `min_length`. Useful
         to ignore the noise of many very short edges.
     weight
-        If not None, weight edges' bearings by this (non-null) edge attribute.
-        For example, if "length" is provided, this will return 1 bearing
-        observation per meter per street (which could result in a very large
-        `bearings` array).
+        If None, return equal weight for each bearing. Otherwise,
+        weight edges' bearings by this (non-null) edge attribute.
+        For example, if "length" is provided, this will weight each bearing
+        observation by the meter length of each street.
 
     Returns
     -------
-    bearings
-        The edge bearings of `Gu`.
+    bearings, weights
+        The edge bearings of `G` and their corresponding weights.
     """
     if projection.is_projected(G.graph["crs"]):  # pragma: no cover
         msg = "Graph must be unprojected to analyze edge bearings."
         raise ValueError(msg)
     bearings = []
+    weights = []
     for u, v, data in G.edges(data=True):
-        # ignore self-loops and any edges below min_length
-        if u != v and data["length"] >= min_length:
-            if weight:
-                # weight edges' bearings by some edge attribute value
-                bearings.extend([data["bearing"]] * int(data[weight]))
-            else:
-                # don't weight bearings, just take one value per edge
-                bearings.append(data["bearing"])
+        if u == v or data["length"] < min_length:
+            # ignore self-loops and any edges below min_length
+            continue
+        bearings.append(data["bearing"])
+        weights.append(data[weight] if weight is not None else 1.0)
```

Comment:
It would be simpler to make this `if` statement the inverse, so you can avoid the `continue` and move lines 217 and 218 under the `if` block.

Reply:
done in https://github.com/gboeing/osmnx/pull/1147/commits/6ca535266967a53abdbf64669f455281f4ef3afb